src/apps/chifra/pkg/cache/cache.go

```diff
@@ -28,17 +29,32 @@ func getCacheAndChainPath(chain string) string {
 func save(chain string, filePath string, content io.Reader) (err error) {
 	cacheDir := getCacheAndChainPath(chain)
 	fullPath := path.Join(cacheDir, filePath)
-	file, err := os.Create(fullPath)
-	if err != nil {
-		return
+
+	var file *os.File
+	if _, err = os.Stat(fullPath); os.IsNotExist(err) {
+		// If file doesn't exist, we don't need a lock
+		file, err = os.Create(fullPath)
```

Comment:
I'm not in love with using GoLang's shorthand functions for opening files. If you look at that code, it just does:

```
OpenFile(name, O_RDWR|O_CREATE|O_TRUNC, 0666)
```

I'd rather have this explicit code in our code so I can see exactly what's happening without having to know how the GoLang code works. In fact, about three months ago, I removed all the "shorthand" code so I can search for `OpenFilename` and see everywhere I'm opening a file of any type.

Reply:
Noted 👍